lib/tasks/doctor.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var defaultDoctor = require('./doctor');
+var Task          = require('../models/task');
+var path          = require('path');
+var fs            = require('fs-extra');
+var defaults      = require('lodash-node/modern/objects/defaults');
+var merge         = require('lodash-node/modern/objects/merge');
+var contains      = require('lodash-node/modern/collections/contains');
+var chalk         = require('chalk');
+var Promise       = require('../ext/promise');
+
+module.exports = Task.extend({
+  init: function() {
+    this.npm = this.npm || require('npm');
+  },
+
+  run: function(options) {
+    var project            = this.project;
+    var isWithinProject    = project.isEmberCLIProject();
+    var defaultDiagnostics = defaultDoctor;
+    var targetDiagnostics  = {};
+
+    // Check to see if the project has it's own diagnostics
+    if (isWithinProject) {
+      targetDiagnostics = this.loadTargetDiagnostics() || {};
+    }
+
+    var checks = merge(targetDiagnostics, defaultDiagnostics, defaults);
+
+    return Promise.all(this.beginChecks(checks, options.skip)).then(function() {
```

Comment:
`this.beginChecks` is misleading as `beingChecks` doesn't depend on any internal state. It could just as easily be its on module. Or it could rely on  internal state rather then having args.